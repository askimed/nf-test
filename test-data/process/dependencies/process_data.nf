process PROCESS_DATA {

  publishDir "${params.outdir}", mode: 'copy'

  input:
  	val name 
    path input

  output:
     path "output.txt", emit: results

  """
  	cp ${input} output.txt
  	echo "hey ${name}!" >> "output.txt"
  """

}
