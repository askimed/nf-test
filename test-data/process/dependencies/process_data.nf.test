nextflow_process {

    name "Test process data"

    script "./process_data.nf"
    process "PROCESS_DATA"

    test("Should use process GENERATE_DATA to generate input data") {

        setup {
            run("GENERATE_DATA") {
                script "./generate_data.nf"
                process {
                    """
                    input[0] = "nf-core"
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = "lukas"
                input[1] = GENERATE_DATA.out.results
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.results).match()
        }
    }

}
