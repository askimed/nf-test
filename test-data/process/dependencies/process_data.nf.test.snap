{
    "Should use process GENERATE_DATA to generate input data": {
        "content": [
            [
                "output.txt:md5,cf78c141ba07d58cba270f3ccc5ee358"
            ]
        ],
        "timestamp": "2023-10-05T12:41:15.52647"
    }
}